File: app/src/main/java/com/denchic45/kts/uivalidator/UIValidator.java
package com.denchic45.kts.uivalidator;

import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.concurrent.atomic.AtomicBoolean;
import java.util.concurrent.atomic.AtomicReference;

public class UIValidator {

    private final List<Validation> validations;

    public UIValidator() {
        validations = new ArrayList<>();
    }

    public UIValidator(List<Validation> validations) {
        this.validations = validations;
    }

    @Contract("_ -> new")
    public static <T> @NotNull UIValidator of(Validation... validations) {
        return new UIValidator(Arrays.asList(validations));
    }

    public void addValidation(Validation validation) {
        validations.add(validation);
    }

    public boolean runValidates() {
        validations.forEach(Validation::validate);
        return validations.stream().allMatch(Validation::validate);
    }

    public void runValidates(Runnable runnable) {
        if (runValidates())
            runnable.run();
    }

//    public boolean isValid() {
//        return validations.stream().allMatch(Validation::isValid);
//    }
}

File: app/src/main/java/com/denchic45/kts/uivalidator/UIValidator.kt
package com.denchic45.kts.uivalidator

import org.jetbrains.annotations.Contract
import java.util.function.Consumer

class UIValidator {
    private val validations: MutableList<Validation>

    constructor() {
        validations = ArrayList()
    }

    constructor(validations: MutableList<Validation>) {
        this.validations = validations
    }

    fun addValidation(validation: Validation) {
        validations.add(validation)
    }

    fun runValidates(): Boolean {
        validations.forEach(Consumer { obj: Validation -> obj.validate() })
        return validations.all { obj: Validation -> obj.validate() }
    }

    fun runValidates(runnable: Runnable) {
        if (runValidates()) runnable.run()
    }

    companion object {
        fun of(vararg validations: Validation): UIValidator {
            return UIValidator(validations.toMutableList())
        }
    }
}